internal/relay/config.go

```diff
@@ -11,13 +12,14 @@ import (
 	neutronapp "github.com/neutron-org/neutron/app"
 )
 
-func GetNeutronChain(logger *zap.Logger, cfg *config.NeutronChainConfig, chainID string) (*relayer.Chain, error) {
+func GetNeutronChain(logger *zap.Logger, cfg *config.NeutronChainConfig, chainID string, keyName string) (*relayer.Chain, error) {
 	provCfg := cosmos.CosmosProviderConfig{
-		Key:            cfg.SignKeyName,
-		ChainID:        chainID,
-		RPCAddr:        cfg.RPCAddr,
-		AccountPrefix:  neutronapp.Bech32MainPrefix,
-		KeyringBackend: cfg.KeyringBackend,
+		Key:           keyName,
+		ChainID:       chainID,
+		RPCAddr:       cfg.RPCAddr,
+		AccountPrefix: neutronapp.Bech32MainPrefix,
+		// we ignore provided keyring here since we're to substitute it later after initialization
```

Comment:
can't we ignore this field right here (just by not setting it at all) and make the substitution later in this very func? i.e. move the substitution from `loadChains` to the end of this func? this would
1. remove this comment;
2. simplify the `loadChains` func;
3. move this part to a more suitable place so the result of `GetNeutronChain` is a ready-to-use chain.